src/configured_yang_push.c

```diff
@@ -0,0 +1,1076 @@
+            sprintf(buf, "%" PRIu32, cyp_data->dampening_period_ms / 10);
+            if (lyd_new_term(cont, NULL, "dampening-period", buf, 0, NULL)) {
+                return SR_ERR_LY;
+            }
+        }
+
+        /* sync-on-start */
+        if (lyd_new_term(cont, NULL, "sync-on-start", cyp_data->sync_on_start ? "true" : "false", 0, NULL)) {
+            return SR_ERR_LY;
+        }
+
+        /* excluded-change* */
+        for (op = 0; op < YP_OP_OPERATION_COUNT; ++op) {
+            if (cyp_data->excluded_change[op]) {
+                if (lyd_new_term(cont, NULL, "excluded-change", yang_push_op2str(op), 0, NULL)) {
+                    return SR_ERR_LY;
+                }
+            }
+        }
+    }
+
+    /* configured-subscription-state */
+    switch (cyp_data->state) {
+    default:
+        strcpy(buf, "valid");
+        break;
+    case CONFIGURED_SUBSCRIPTION_STATE_CONCLUDED:
+        strcpy(buf, "concluded");
+        break;
+    }
+
+    /* configured-subscription-state */
+    if (lyd_new_term(subscription, NULL, "configured-subscription-state", buf, 0, NULL)) {
+        return SR_ERR_LY;
+    }
+
+    return SR_ERR_OK;
+}
+
+uint32_t
+cfg_yang_push_oper_receiver_excluded(struct np2srv_sub_ntf *sub)
+{
+    struct cfg_yang_push_data *cyp_data = sub->data;
+    uint32_t i, excluded_count = 0, filtered_out;
+    int r;
+
+    if (!cyp_data->periodic) {
+        /* excluded-event-records */
+        for (i = 0; i < ATOMIC_LOAD_RELAXED(sub->sub_id_count); ++i) {
+            /* get filter-out count for the subscription */
+            r = sr_module_change_sub_get_info(np2srv.sr_cfg_data_sub, sub->sub_ids[i], NULL, NULL, NULL, &filtered_out);
+            if (r != SR_ERR_OK) {
+                return 0;
+            }
+            excluded_count += filtered_out;
+        }
+
+        /* add excluded op */
+        excluded_count += ATOMIC_LOAD_RELAXED(cyp_data->excluded_op_count);
+    } /* else always zero */
+
+    return excluded_count;
+}
+
+void
+cfg_yang_push_terminate_async(struct np2srv_sub_ntf *sub)
+{
+    struct cfg_yang_push_data *cyp_data = sub->data;
+    struct itimerspec tspec = {0};
+
+    /* disarm all timers */
+    if (cyp_data->periodic) {
+        timer_settime(cyp_data->update_timer, TIMER_ABSTIME, &tspec, NULL);
+    } else {
+        if (cyp_data->dampening_period_ms) {
+            timer_settime(cyp_data->damp_timer, TIMER_ABSTIME, &tspec, NULL);
+        }
+    }
+
+    if (cyp_data->stop_timer) {
+        const struct ly_ctx *ly_ctx;
+        struct lyd_node *ly_ntf;
+        char buf[26];
+
+        timer_settime(cyp_data->stop_timer, TIMER_ABSTIME, &tspec, NULL);
+
+        ly_ctx = sr_acquire_context(np2srv.sr_conn);
+        sr_release_context(np2srv.sr_conn);
+
+        sprintf(buf, "%" PRIu32, sub->nc_sub_id);
+        lyd_new_path(NULL, ly_ctx, "/ietf-subscribed-notifications:subscription-completed/id",
+                     buf, 0, &ly_ntf);
```

Comment:
In our coding style (which will be enforced by `uncrustify`) we do not align parameters with the function parentheses but simply continue with double indent on the next line.